ts/features/bonus/bpd/screens/optInPaymentMethods/OptInPaymentMethodsThankYouKeepMethodsScreen.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect } from "react";
+import { useDispatch } from "react-redux";
+import * as pot from "italia-ts-commons/lib/pot";
+import ThankYouSuccessComponent from "../../components/optInPaymentMethods/ThankYouSuccessComponent";
+import { useIOSelector } from "../../../../../store/hooks";
+import { optInPaymentMethodsCompleted } from "../../store/actions/optInPaymentMethods";
+import { showToast } from "../../../../../utils/showToast";
+import I18n from "../../../../../i18n";
+import { optInStatusSelector } from "../../store/reducers/details/activation";
+import { LoadingErrorComponent } from "../../../bonusVacanze/components/loadingErrorScreen/LoadingErrorComponent";
+import { bpdUpdateOptInStatusMethod } from "../../store/actions/onboarding";
+import { CitizenOptInStatusEnum } from "../../../../../../definitions/bpd/citizen_v2/CitizenOptInStatus";
+
+const OptInPaymentMethodsThankYouKeepMethodsScreen = () => {
+  const dispatch = useDispatch();
+  const optInStatus = useIOSelector(optInStatusSelector);
+
+  useEffect(() => {
+    // dispatch saga that delete payment methods and update opt-in field
+    dispatch(
+      bpdUpdateOptInStatusMethod.request(CitizenOptInStatusEnum.ACCEPTED)
+    );
+  }, [dispatch]);
+
+  const isOptInStatusLoading = pot.fold(
+    optInStatus,
+    () => true,
+    () => true,
+    _ => true,
+    _ => false,
+    _ => false,
+    _ => true,
+    _ => true,
+    _ => false
+  );
+
+  // if optInStatus is on loading (or undefined) state show loading component
+  if (isOptInStatusLoading) {
+    return (
+      <LoadingErrorComponent
+        isLoading={true}
+        loadingCaption={I18n.t("global.remoteStates.loading")}
+        onRetry={() => true}
+        testID={"loadingComponent"}
+      />
+    );
+  }
+
+  // if the opt-in choice fails complete the workunit and show an error toast to the user
+  if (pot.isError(optInStatus)) {
+    showToast(I18n.t("bonus.bpd.optInPaymentMethods.thankYouPage.toast"));
+    dispatch(optInPaymentMethodsCompleted());
+    return null;
+  }
```

Comment:
Same issue of [OptInPaymentMethodsThankYouDeleteMethodsScreen.tsx](https://github.com/pagopa/io-app/pull/3763/files#diff-05b923a291a4a3df0f1e8659e39ad452b2b27889b5a9dbf0cae07711ff40573b)

Reply:
Refactored here [fdfc7d5](https://github.com/pagopa/io-app/pull/3763/commits/fdfc7d5cc4c18174d0d0b40604a72f76fa6cfe31)